comp/dogstatsd/server/server_integration_test.go

```diff
@@ -0,0 +1,173 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build test
+
+package server
+
+import (
+	"context"
+	"net"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/listeners"
+)
+
+func TestUDPForward(t *testing.T) {
+	cfg := make(map[string]interface{})
+
+	pc, err := net.ListenPacket("udp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	pcHost, pcPort, err := net.SplitHostPort(pc.LocalAddr().String())
+	require.NoError(t, err)
+
+	// Setup UDP server to forward to
+	cfg["statsd_forward_port"] = pcPort
+	cfg["statsd_forward_host"] = pcHost
+
+	// Setup dogstatsd server
+	cfg["dogstatsd_port"] = listeners.RandomPortName
+
+	deps := fulfillDepsWithConfigOverride(t, cfg)
+
+	defer pc.Close()
+
+	requireStart(t, deps.Server)
+
+	conn, err := net.Dial("udp", deps.Server.UDPLocalAddr())
+	require.NoError(t, err)
+	require.NotNil(t, conn)
+	defer conn.Close()
+
+	// Check if message is forwarded
+	_, err = conn.Write(defaultMetricInput)
+	require.NoError(t, err, "cannot write to DSD socket")
+
+	_ = pc.SetReadDeadline(time.Now().Add(4 * time.Second))
+
+	buffer := make([]byte, len(defaultMetricInput))
+	_, _, err = pc.ReadFrom(buffer)
+	require.NoError(t, err)
+
+	assert.Equal(t, defaultMetricInput, buffer)
+}
+
+func TestUDPConn(t *testing.T) {
```

Comment:
TestUDPConn and TestUDSConn both retain the potential for instability on the mac runner. I've significantly reduced the scope of the original TestUDSReceiver and TestUDPReceive methods that ran this flow in the hopes that  I've reduced the likelihood of a flake, focusing on just a single metric/event/service case running from socket to aggregator. Will continue to poke at the flakes to see if I can improve this any further beyond just spiking up timeouts, very open to other inputs.